ui-participant/src/participant/ParticipantProfile.tsx

```diff
@@ -0,0 +1,170 @@
+            }/>
+          </ProfileRow>
+          <ProfileRow title={'Birthday'} onEdit={() => setShowEditFieldModal('birthDate')}>
+            <ProfileTextRow text={
+              profile.birthDate && dateToDefaultString(profile.birthDate)
+            }/>
+          </ProfileRow>
+        </ProfileCard>
+
+        <ProfileCard title="Mailing Address">
+          <ProfileRow title={'Primary Address'} onEdit={() => setShowEditFieldModal('mailingAddress')}>
+            <ReadOnlyAddress address={profile.mailingAddress}/>
+          </ProfileRow>
+        </ProfileCard>
+
+        <ProfileCard title="Communication Preferences">
+          <ProfileRow title={'Contact Email'} onEdit={() => setShowEditFieldModal('contactEmail')}>
+            <ProfileTextRow text={profile.contactEmail}/>
+          </ProfileRow>
+          <ProfileRow title={'Phone Number'} onEdit={() => setShowEditFieldModal('phoneNumber')}>
+            <ProfileTextRow text={profile.phoneNumber}/>
+          </ProfileRow>
+          <ProfileRow title={'Notifications'} onEdit={() => setShowEditFieldModal('doNotEmail')}>
+            <ProfileTextRow text={profile.doNotEmail ? 'Off' : 'On'}/>
+          </ProfileRow>
+          <ProfileRow title={'Do Not Solicit'} onEdit={() => setShowEditFieldModal('doNotEmailSolicit')}>
+            <ProfileTextRow text={profile.doNotEmailSolicit ? 'On' : 'Off'}/>
+          </ProfileRow>
+        </ProfileCard>
+
+        {/*Edit modals*/}
+        <EditParticipantProfileFieldModals
+          profile={profile}
+          showFieldModal={showEditFieldModal}
+          dismissModal={() => setShowEditFieldModal(undefined)}
+          save={save}/>
+      </div>
+    </div>
+  </div>
+}
+
+function ProfileCard({ title, children }: { title: string, children: React.ReactNode }) {
+  return <div className="card mb-3">
+    <div className="card-body p-4">
+      <h2 className="fw-bold pb-3">{title}</h2>
+      {children}
+    </div>
+  </div>
+}
+
+const ProfileTextRow = ({ text }: { text: string | undefined }) => {
+  return (!isEmpty(text)
+    ? <p className="m-0">{text}</p>
+    : <p className="m-0 fst-italic text-secondary">Not provided</p>
+  )
+}
+
+const ReadOnlyAddress = ({ address }: { address: MailingAddress | undefined }) => {
+  if (
+    !address || (
+      isEmpty(address.city)
+      && isEmpty(address.street1)
+      && isEmpty(address.street2)
+      && isEmpty(address.country)
+      && isEmpty(address.postalCode)
+      && isEmpty(address.state)
+    )) {
+    return <ProfileTextRow text={undefined}/>
+  }
+
+  return <>
+    {address.street1 && <ProfileTextRow text={address.street1}/>}
+    {address.street2 && <ProfileTextRow text={address.street2}/>}
+    {(address.city || address.postalCode || address.state) &&
+        <ProfileTextRow text={`${address.city} ${address.state} ${address.postalCode}`.split(/\s+/).join(' ')}/>}
+    {address.country && <ProfileTextRow text={address.country}/>}
+  </>
+}
+
+const Bar = () => {
```

Comment:
Always throws me off when there's a legitimate use of "foo" or "bar" left in code. :) Rename to `HorizontalBar` or something?